bookkeeper-server/src/main/java/org/apache/bookkeeper/discover/BookieServiceInfo.java

```diff
@@ -0,0 +1,178 @@
+
+    public BookieServiceInfo(Map<String, String> properties, List<Endpoint> endpoints) {
+        this.properties = Collections.unmodifiableMap(properties);
+        this.endpoints = Collections.unmodifiableList(endpoints);
+    }
+
+    public BookieServiceInfo() {
+    }
+
+    /**
+     * Unmodifiable map with bookie wide information.
+     *
+     * @return the map
+     */
+    public Map<String, String> getProperties() {
+        return properties;
+    }
+
+    /**
+     * Unmodifieable structure with the list of exposed endpoints.
+     *
+     * @return the list.
+     */
+    public List<Endpoint> getEndpoints() {
+        return endpoints;
+    }
+
+    public void setProperties(Map<String, String> properties) {
+        this.properties = properties;
+    }
+
+    public void setEndpoints(List<Endpoint> endpoints) {
+        this.endpoints = endpoints;
+    }
+
+    /**
+     * Information about an endpoint.
+     */
+    public static final class Endpoint {
+
+        private String id;
+        private int port;
+        private String host;
+        private String protocol;
+        private List<String> auth;
+        private List<String> extensions;
+
+        public Endpoint(String id, int port, String host, String protocol, List<String> auth, List<String> extensions) {
+            this.id = id;
+            this.port = port;
+            this.host = host;
+            this.protocol = protocol;
+            this.auth = auth;
+            this.extensions = extensions;
+        }
+
+        public Endpoint() {
+        }
+
+        public String getId() {
+            return id;
+        }
+
+        public void setId(String id) {
+            this.id = id;
+        }
+
+        public int getPort() {
+            return port;
+        }
+
+        public void setPort(int port) {
+            this.port = port;
+        }
+
+        public String getHost() {
+            return host;
+        }
+
+        public void setHost(String host) {
+            this.host = host;
+        }
+
+        public String getProtocol() {
+            return protocol;
+        }
+
+        public void setProtocol(String protocol) {
+            this.protocol = protocol;
+        }
+
+        public List<String> getAuth() {
+            return auth;
+        }
+
+        public void setAuth(List<String> auth) {
+            this.auth = auth;
+        }
+
+        public List<String> getExtensions() {
+            return extensions;
+        }
+
+        public void setExtensions(List<String> extensions) {
+            this.extensions = extensions;
+        }
+
+        @Override
+        public String toString() {
+            return "EndpointInfo{" + "id=" + id + ", port=" + port + ", host=" + host + ", protocol=" + protocol + ", "
+                    + "auth=" + auth + ", extensions=" + extensions + '}';
+        }
+
+    }
+
+    public static BookieServiceInfo buildLegacyBookieServiceInfo(String bookieId) throws UnknownHostException {
```

Comment:
javadoc

Reply:
what about moving this function to a BpploeServiceInfoUtil class ?

please also add unit tests